connectors/gitlab/element-templates/gitlab-connector.json

```
@@ -852,66 +889,562 @@
+        "name": "body"
+      },
+      "condition": {
+        "property": "operation",
+        "oneOf": [
+          "createRepositoryBranch"
+        ]
+      }
+    },
+    {
+      "label": "Branch name",
+      "description": "Name of the new branch to create the file in. The commit is added to this branch",
+      "group": "operation",
+      "type": "String",
+      "feel": "optional",
+      "binding": {
+        "type": "zeebe:input",
+        "name": "branchName"
+      },
+      "constraints": {
+        "notEmpty": true
+      },
+      "condition": {
+        "property": "operation",
+        "oneOf": [
+          "createRepositoryFile"
+        ]
+      }
+    },
+    {
+      "label": "Commit message",
+      "group": "operation",
+      "type": "String",
+      "feel": "optional",
+      "binding": {
+        "type": "zeebe:input",
+        "name": "commitMessage"
+      },
+      "constraints": {
+        "notEmpty": true
+      },
+      "condition": {
+        "property": "operation",
+        "oneOf": [
+          "createRepositoryFile"
+        ]
+      }
+    },
+    {
+      "label": "Content",
+      "group": "operation",
+      "type": "Text",
+      "feel": "optional",
+      "binding": {
+        "type": "zeebe:input",
+        "name": "content"
+      },
+      "constraints": {
+        "notEmpty": true
+      },
+      "condition": {
+        "property": "operation",
+        "oneOf": [
+          "createRepositoryFile"
+        ]
+      }
+    },
+    {
+      "label": "File path",
+      "description": "URL-encoded full path to new file. For example, lib%2Fclass%2Erb",
+      "group": "operation",
+      "type": "String",
+      "feel": "optional",
+      "binding": {
+        "type": "zeebe:input",
+        "name": "filePath"
+      },
+      "constraints": {
+        "notEmpty": true
+      },
+      "condition": {
+        "property": "operation",
+        "oneOf": [
+          "createRepositoryFile"
+        ]
+      }
+    },
+    {
+      "label": "Author email",
+      "description": "The commit author's email address",
+      "group": "operation",
+      "type": "String",
+      "feel": "optional",
+      "optional": true,
+      "binding": {
+        "type": "zeebe:input",
+        "name": "authorEmail"
+      },
+      "condition": {
+        "property": "operation",
+        "oneOf": [
+          "createRepositoryFile"
+        ]
+      }
+    },
+    {
+      "label": "Author name",
+      "description": "The commit author's name",
+      "group": "operation",
+      "type": "String",
+      "feel": "optional",
+      "optional": true,
+      "binding": {
+        "type": "zeebe:input",
+        "name": "authorName"
+      },
+      "condition": {
+        "property": "operation",
+        "oneOf": [
+          "createRepositoryFile"
+        ]
+      }
+    },
+    {
+      "label": "Encoding",
+      "description": "Change encoding to <code>base64</code>. Default is <code>text</code>",
```

```suggestion
      "description": "Change encoding to <code>base64</code>. Gitlab's default is <code>text</code>",
```

I recognize this description from [Gitlab's API docs](https://docs.gitlab.com/ee/api/repository_files.html#create-new-file-in-repository), which is good to use.

My intention here is to avoid users thinking this element template (or the Connector behind it) is setting this default.